packages/astro/src/core/config/schema.ts

```diff
@@ -621,7 +621,10 @@ export const AstroConfigSchema = z.object({
 									}),
 								),
 							localFontFamilySchema,
-							commonFontFamilySchema,
+							commonFontFamilySchema.transform((family) => ({
+								provider: GOOGLE_PROVIDER_NAME,
+								...family,
+							})),
```

Comment:
I would avoid a `transform` inside for our configuration. I used `.transform` in the past, which is the side effect of passing the transformed configuration **to the integrations**. This means they will have incorrect types, and they might be able to delete our defaults.

I experienced this inconvenience when I implemented i18n. I suggest injecting these defaults after the integrations run their hooks.

Reply:
You mean the transforms don't run after `updateConfig()` is called? That would be bad indeed

Reply:
Yes. For v5, I wanted to break this, provide the unparsed/unvalidated configuration to the integrations, and eventually run Zod parsing afterward.

Unfortunately the architecture of Astro doesn't allow it, Astro requires the transformed/parsed values before it could run the integration hooks.